lib/models/ReplicationConfiguration.js

```diff
@@ -301,7 +301,8 @@ class ReplicationConfiguration {
             const endpoint = replicationEndpoints.find(endpoint =>
                 endpoint.site === storageClass);
             if (endpoint) {
-                this._hasScalityDestination = endpoint.type ? null : true;
+                this._hasScalityDestination =
+                    !this._hasScalityDestination && endpoint.type ? null : true;
```

Comment:
I have the same concern.

Reply:
This is fixing the case where `this._hasScalityDestination` had been set to true in a prior loop here, so that it is not reset to `null` on any following endpoints. Will try to make this more readable.

Reply:
I have an alternative but am finding it to be just as difficult to follow since it has to factor in all the rules, and the case of a previous `this._hasScalityDestination` being overwritten still applies. I will push a comment here to make this more clear.